src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/SafeTransactionPreview.tsx

```diff
@@ -119,84 +138,173 @@ interface Props {
+                    /* Need casting as `formatValues` func used in Heading
+                            doesn't have correct types (coming from react-intl) */
+                  ) as PrimitiveType,
+                }}
               />
+              <Button
+                className={styles.tabButton}
+                onClick={() => handleTabToggle(index)}
+              >
+                <Icon
+                  name="caret-up-small"
+                  className={styles.toggleTabIcon}
+                  title={MSG.toggleTransaction}
+                  titleValues={{
+                    tabToggleStatus: transactionTabStatus[index],
+                  }}
+                />
+              </Button>
             </div>
           )}
-          <DetailsItem
-            label={MSG.safe}
-            value={
-              <AddressDetailsView
-                item={(values.safe as never) as AnyUser}
-                isSafeItem
-              />
-            }
-          />
-          {values.transactions[0].transactionType !==
-            TransactionTypes.CONTRACT_INTERACTION && (
-            <DetailsItem
-              label={MSG.to}
-              value={
-                <AddressDetailsView
-                  item={(values.transactions[0].recipient as never) as AnyUser}
-                  isSafeItem={false}
+          <div
+            className={classnames({
+              [styles.tabContentClosed]:
+                values.transactions.length > 1 && !transactionTabStatus[index],
+            })}
+          >
+            <DialogSection>
+              <div className={styles.transactionDetailsSection}>
+                {values.transactions[index].transactionType !== '' &&
+                  values.transactions.length === 1 && (
+                    <div className={styles.transactionTitle}>
+                      1.{' '}
+                      <FormattedMessage
+                        {...ConstantsMSG[
+                          values.transactions[index].transactionType
+                        ]}
+                      />
+                    </div>
+                  )}
+                <DetailsItem
+                  label={MSG.safe}
+                  value={
+                    <AddressDetailsView
+                      item={(values.safe as never) as AnyUser}
+                      isSafeItem
+                    />
+                  }
                 />
-              }
-            />
-          )}
-          {transactionDetails?.map(({ key, label, value }) => (
-            <DetailsItem
-              key={key}
-              label={label}
-              value={value(values.transactions[0][key], token)}
-            />
-          ))}
+                {values.transactions[index].transactionType !==
+                  TransactionTypes.CONTRACT_INTERACTION && (
+                  <DetailsItem
+                    label={MSG.to}
+                    value={
+                      <AddressDetailsView
```

Comment:
This component needs to be refactored (not in this PR). Like SingleUserPicker, <AddressDetailsView> should be generic. 
You're having to make lots of casts be able to pretend to be a user. But the component should handle, User, Safe, NFT.... etc.

Also this component never displays an avatar for me. Perhaps this is due to my dev env. I check and saw that the avatars are shown in other peoples screen captures.
<img width="532" alt="Screenshot 2022-08-04 at 15 54 31" src="https://user-images.githubusercontent.com/582700/182864390-3483fa85-7618-4324-bdee-b8a8d0b93276.png">

Reply:
> This component needs to be refactored (not in this PR). Like SingleUserPicker, should be generic. You're having to make lots of casts be able to pretend to be a user. But the component should handle, User, Safe, NFT.... etc.
I created a new task for the refactoring [here](https://github.com/JoinColony/colonyDapp/issues/3730).

> Also this component never displays an avatar for me. Perhaps this is due to my dev env. I check and saw that the avatars are shown in other peoples screen captures. <img alt="Screenshot 2022-08-04 at 15 54 31" width="532" src="https://user-images.githubusercontent.com/582700/182864390-3483fa85-7618-4324-bdee-b8a8d0b93276.png">
Can you please check again? It seems to be working for everyone else.